src/engine.js

```diff
@@ -191,11 +192,13 @@ class Engine extends EventEmitter {
       return rule.evaluate(almanac).then((ruleResult) => {
         debug(`engine::run ruleResult:${ruleResult.result}`)
         if (ruleResult.result) {
-          this.emit('success', rule.event, almanac, ruleResult)
-          this.emit(rule.event.type, rule.event.params, almanac, ruleResult)
-          almanac.factValue('success-events', { event: rule.event })
+          almanac.factValue('success-events', { event: ruleResult.event })
+          return Promise.all([
+            this.emitAsync('success', ruleResult.event, almanac, ruleResult),
+            this.emitAsync(ruleResult.event.type, ruleResult.event.params, almanac, ruleResult)
```

Comment:
In the current implementation `success` & `ruleResult.event.type` is triggered sequentially. This change is triggering them concurrently. Not a problem for us but just wanted to check if it is intentional.

Reply:
another good call out! I can't see a good reason reason to deviate from the previous behavior; I've updated this to fire these in series, like it did before.